contracts/its-stacks-translator/src/contract/its_clarity_bytes_to_hub_message.rs

```diff
@@ -0,0 +1,422 @@
+use std::str::FromStr;
+
+use clarity::vm::representations::ClarityName;
+use clarity::vm::types::{
+    BufferLength, SequenceSubtype, StringSubtype, TupleData, TupleTypeSignature, TypeSignature,
+    Value,
+};
+use cosmwasm_std::Uint128;
+use cw_storage_plus::KeyDeserialize;
+use interchain_token_service_std::{
+    DeployInterchainToken, HubMessage, InterchainTransfer, Message, TokenId,
+};
+use router_api::ChainNameRaw;
+use stacks_common::codec::StacksMessageCodec;
+
+use crate::contract::its_hub_message_to_clarity_bytes::{
+    CLARITY_NAME_AMOUNT, CLARITY_NAME_DATA, CLARITY_NAME_DECIMALS,
+    CLARITY_NAME_DESTINATION_ADDRESS, CLARITY_NAME_NAME, CLARITY_NAME_SOURCE_ADDRESS,
+    CLARITY_NAME_SYMBOL, CLARITY_NAME_TOKEN_ID, CLARITY_NAME_TYPE,
+    MESSAGE_TYPE_DEPLOY_INTERCHAIN_TOKEN, MESSAGE_TYPE_INTERCHAIN_TRANSFER,
+};
+use crate::error::ContractError;
+
+const MESSAGE_TYPE_SEND_TO_HUB: u128 = 3;
+
+const CLARITY_NAME_DESTINATION_CHAIN: &str = "destination-chain";
+const CLARITY_NAME_PAYLOAD: &str = "payload";
+const CLARITY_NAME_MINTER: &str = "minter";
+const CLARITY_SIZE_DESTINATION_CHAIN: u32 = 19;
```

Comment:
shouldn't this be 20?

Reply:
Max chain name length is 19 according to the integration document:
https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/gateway/INTEGRATION.md#limits

That is how it was implemented on Stacks:
https://github.com/Trust-Machines/axelar_stacks_bridge/blob/master/INTEGRATION.md#gateway
https://github.com/Trust-Machines/axelar_stacks_bridge/blob/master/contracts/interchain-token-service-impl.clar#L583

Reply:
yes the max length is 20, we should change this, else if we end up with a 20 character chain, transfers will fail